dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java

```diff
@@ -156,13 +162,87 @@ private void getEnrollments( EventQueryParams params, Grid grid, String sql, boo
 
             grid.addRow();
 
+            // columnOffset is synchronization aid for <<grid headers>> and <<rowSet columns>> indexes.
+            // The amount of headers must not match to amount of columns due the additional ones
+            // describing the repeating of repeatable stage.
+            int columnOffset = 0;
+
             for ( int i = 0; i < grid.getHeaders().size(); ++i )
             {
-                addGridValue( grid, grid.getHeaders().get( i ), i + 1, rowSet, params );
+                addGridValue( grid, grid.getHeaders().get( i ), i + 1 + columnOffset, rowSet, params );
+
+                if ( addValueMetaInfo( grid, rowSet, grid.getHeaders().get( i ).getName() ) )
+                {
+                    ++columnOffset;
+                }
             }
         }
     }
 
+    /**
+     * Add value meta info into the grid. Value meta info is information about
+     * origin of the repeatable stage value.
+     *
+     * @param grid the {@link Grid}.
+     * @param rowSet the {@link SqlRowSet}.
+     * @param columnName the {@link String}.
+     * @return true when ValueMetaInfo added
+     */
+    private boolean addValueMetaInfo( Grid grid, SqlRowSet rowSet, String columnName )
+    {
+        int gridRowIndex = grid.getRows().size() - 1;
+
+        Optional<String> valueMetaInfoColumnName = Arrays.stream( rowSet.getMetaData().getColumnNames() )
+            .filter( (columnName + ".exists")::equalsIgnoreCase )
+            .findFirst();
+
+        if ( valueMetaInfoColumnName.isPresent() )
+        {
+            try
+            {
+                boolean isDefined = rowSet.getBoolean( valueMetaInfoColumnName.get() );
+
+                boolean isSet = rowSet.getObject( columnName ) != null;
+
+                Map<Integer, Map<String, Object>> rowContext = grid.getRowContext();
+
+                Map<String, Object> row = rowContext.get( gridRowIndex );
+
+                if ( row == null )
+                {
+                    row = new HashMap<>();
+                }
+
+                Map<String, RepeatableStageValueStatus> colValueType = new HashMap<>();
+
+                colValueType.put( "type", getGridValueStatus( isDefined, isSet ) );
+
+                row.put( columnName, colValueType );
+
+                rowContext.put( gridRowIndex, row );
+
+                return true;
+            }
+            catch ( InvalidResultSetAccessException ignored )
+            {
+                // when .exists extension of column name does not indicate boolean flag,
+                // value will not be added and method returns false
+            }
+        }
+
+        return false;
+    }
+
+    private RepeatableStageValueStatus getGridValueStatus( boolean isDefined, boolean isSet )
```

Comment:
This method could be moved to the Enum, so you could have
```
RepeatableStageValueStatus.of (boolean isDefined, boolean isSet)
```

Reply:
Done